### ISSUE TITLE
Self-Hosted VCS provider

### ISSUE THREAD

**Issue by alfespa17** [AzBuilder/terrakube#590]
### Feature description 💡

Add support to use self hosted VCS provider like Github Enterprise, Gitlab Server, Bitbucket Server.

### Anything else?

N/A

**Comment by alfespa17** [comment]
I already did the necessary changes in the code, but I don't have a self-hosted VCS if anyone is interested in test this feature please leave a comment.

**Comment by btpier** [comment]
@alfespa17  This is great! Thank you for the quick work. We would be happy to test it against our self-hosted gitlab.

**Comment by alfespa17** [comment]
Sounds great @btpier, maybe you can install the normal terrakube version and I will create a new beta release so you can test the containers with the change that I did a couple of days ago. Once you have the "normal" terrakube version working in your environment you will only need to override the container version so you can test with your self hosted gitlab

**Comment by btpier** [comment]
@alfespa17 We have the normal version running in our Openshift environment. Having an issue with S3 state storage that we haven't figured out but we have the rest of the system up and working.

**Comment by alfespa17** [comment]
Hello @btpier 

I created a new release so you can help me testing that feature [here](https://github.com/AzBuilder/terrakube/actions/runs/6857581637).

**MAKE SURE TO TEST THIS WITH A TEST DATABASE, IT CONTAINS SOME CHANGES THAT WILL CHANGE THE VCS TABLE**

You could test it overriding the version parameter in the helm chart like this 
```
api:
  version: "2.18.0-vcs-endpoint"

ui:
  version: "2.18.0-vcs-endpoint"
```

Let me know if you face any issue and thank you for helping me testing this new feature

**Comment by btpier** [comment]
@alfespa17 The new version is working well with our self hosted gitlab! A minor UI detail: In the VCS settings, the API URL is still  https://gitlab.com/api/v4 instead of our self-hosted URL.

**Comment by alfespa17** [comment]
Hello @btpier I am glad that is working, could you share an screenshot? Not sure what VCS setting you are talking about

**Comment by jcanizalez** [comment]
Hi @btpier,

I fixed the hardcoded API URL. We have improved the UI for GitLab EE and CE VCS Setup and we have released this feature in version 2.18.0. [Here is the documentation](https://docs.terrakube.io/user-guide/vcs-providers/gitlab-ee-and-ce) that explains how to use this feature.

Thank you for your help.
